nvflare/private/fed/client/client_engine_executor_spec.py

```diff
@@ -63,6 +64,17 @@ def get_widget(self, widget_id: str) -> Widget:
     def get_all_components(self) -> dict:
         pass
 
+    def validate_clients(self, target_names) -> ([], []):
+        """To validate the target names.
+
+        Args:
+            target_names: input target names
+
+        Returns: (valid_names, invalid_names)
+
+        """
+        pass
+
```

Comment:
What does it mean to have `validate_clients` in `client_engine_executor_spec`?

Should we rename to `validate_targets` or something else?

Reply:
extracted the validate_targets to EngineSpec. Both server and client engine require this method.